### ISSUE TITLE
Support for table functions

### ISSUE THREAD

**Issue by buremba** [trinodb/trino#9228]
In some SQL databases (Postgresql, Oracle, etc.), it's valid to call functions in `FROM`. Here is an example query:

```
SELECT count(*) from generate_series(0, 100)
```

The feature is useful for generating random data or dynamically returning datasets from Java UDFs.

My understanding is that we need to implement the following features to support table functions:

1. Support for calling functions in `Relation` node of SQL dialect.
2. Support for returning `RowType` for scalar functions.

If there is interest, I can implement `1` as a start, what do you think?

**Comment by buremba** [comment]
Closing as this is duplicate of #1839 .

**Comment by findepi** [comment]
For this specific example, you can write the following in Trino

```sql
SELECT count(*) FROM UNNEST(sequence(0, 100));
```

> Support for returning RowType for scalar functions.

A scalar function can return a Row today.

> Support for calling functions in Relation node of SQL dialect.

Do you mean feature called "Polymorphic Table Functions"?
I think you already found https://github.com/trinodb/trino/issues/1839

**Comment by buremba** [comment]
Yes, that was what I'm looking for. :) I searched for the keyword `table functions` but missed that issue so wanted to create one. @ebyhr guided me so closed the issue in favor of #1839.
